python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -567,6 +676,65 @@ def calculateMetrics(self, difference):
         detNegPix &= badPix
         self.metadata.add("nBadPixelsDetectedPositive", np.sum(detPosPix))
         self.metadata.add("nBadPixelsDetectedNegative", np.sum(detNegPix))
+        metricsMaskPlanes = []
+        for maskPlane in self.config.metricsMaskPlanes:
+            try:
+                self.metadata.add("%s_mask_fraction"%maskPlane.lower(), evaluateMaskFraction(mask, maskPlane))
+                metricsMaskPlanes.append(maskPlane)
+            except InvalidParameterError:
+                self.metadata.add("%s_mask_fraction"%maskPlane.lower(), -1)
+                self.log.info("Unable to calculate metrics for mask plane %s: not in image"%maskPlane)
+
+        if self.config.doWriteMetrics:
+            summaryMetrics = afwTable.SourceCatalog(self.metricSchema)
+            summaryMetrics.getTable().setIdFactory(idFactory)
+            self.addSkySources(summaryMetrics, science.mask, difference.info.id,
+                               subtask=self.metricSources)
+            for src in summaryMetrics:
+                self._evaluateLocalMetric(src, diaSources, science, matchedTemplate, difference,
+                                          metricsMaskPlanes=metricsMaskPlanes)
+
+            return summaryMetrics.asAstropy()
+
+    def _evaluateLocalMetric(self, src, diaSources, science, matchedTemplate, difference,
+                             metricsMaskPlanes, size=100):
+        bbox = src.getFootprint().getBBox()
+        pix = bbox.getCenter()
+        src.set('science_psfSize', getPsfFwhm(science.psf, position=pix))
+        src.set('template_psfSize', getPsfFwhm(matchedTemplate.psf, position=pix))
+
+        bbox.grow(size)
+        bbox = bbox.clippedTo(science.getBBox())
+        nPix = bbox.getArea()
+        pixScale = science.wcs.getPixelScale()
+        area = nPix*pixScale.asDegrees()**2
+        peak = src.getFootprint().getPeaks()[0]
+        src.set('x', peak['i_x'])
+        src.set('y', peak['i_y'])
+        src.setCoord(science.wcs.pixelToSky(peak['i_x'], peak['i_y']))
+        selectSources = diaSources[bbox.contains(diaSources.getX(), diaSources.getY())]
+        if self.config.doSkySources:
+            selectSources = selectSources[~selectSources["sky_source"]]
+        sourceDensity = len(selectSources)/area
+        dipoleSources = selectSources[selectSources["ip_diffim_DipoleFit_flag_classification"]]
+        dipoleDensity = len(dipoleSources)/area
+        if dipoleSources:
+            meanDipoleOrientation = _angleMean(dipoleSources["ip_diffim_DipoleFit_orientation"])
```

Comment:
The docstring above said this was relative to the parallactic angle--I don't see that computation here, am I missing something?

Reply:
I originally thought it should be relative to parallactic angle, but then decided to simplify the calculation. I'll update the docstring.